ShortURLBuilderProvider/Sources/ShortURLBuilderProvider/Credentials/CredentialsManager.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+public enum CredentialType: String, CaseIterable {
+    case bitlyToken = "BitlyToken"
+    case cuttlyApiKey = "CuttlyApiKey"
+
+    public var displayName: String {
+        switch self {
+        case .bitlyToken:
+            return "Bitly Token"
+        case .cuttlyApiKey:
+            return "Cuttly API Key"
+        }
+    }
+}
+
+@MainActor
+public final class CredentialsManager: ObservableObject {
+    public static let shared = CredentialsManager()
```

Comment:
Technically, no. But the thing is, if I don't do this, I will need to manage instances manually, i.e., each view creates its own instance.

IMHO, it would be a bit inefficient and tedious.

Also, there is a slight concern about the potentially inconsistent state across the app.

And probably some drawbacks like each instance doing its own loadCredentials() call
and memory overhead from multiple instances?

But as a newbie, I will take any suggestion from you 🙂 

Let me know what you think.

Reply:
I agree there is a potential issue with inconsistent state if we create multiple instances. However, since the manager is only responsible for providing functions to save and retrieve data from the Keychain without storing any shared sources, making it a singleton doesn't seem necessary to me.

But it’s a fair point! If so, we could consider using `EnvironmentObject`, a SwiftUI property wrapper that enables data sharing from parent to child views. This would eliminate the need for a shared property.
What do you think?

```swift
public struct CredentialsView: View {
    @EnvironmentObject private var credentialsManager: CredentialsManager
}
```

Reply:
Oki, let me take a look 👀

Reply:
Alright, this is my first time seeing this in Swift. I have tried my best...

Reply:
don't forget I am not a ios engineer 😂

Reply:
Actually you've done a pretty good job as this is the first PR! Just tried to give you more tips in SwiftUI 😛